### ISSUE TITLE
xla bf16 compute accuracy issues

### ISSUE THREAD

**Issue by mars1248** [pytorch/xla#8367]
## ❓ Questions and Help
I built a simple demo to reproduce this problem.
test.py
```
import torch
import torch_xla.core.xla_model as xm
import math
input_dict = torch.load('ut_input_dict.pth')
attention_scores = torch.matmul(input_dict['query_layer'].to(xm.xla_device()), input_dict['key_layer'].to(xm.xla_device()).transpose(-1, -2))
print(attention_scores)
```
my execute command is `XLA_IR_DEBUG=1 XLA_FLAGS="--xla_dump_to=./debug_lynx_hlo --xla_gpu_cudnn_gemm_fusion_level=3 --xla_gpu_enable_triton_gemm=false" python test.py`
above `attention_scores` != torch.matmul(input_dict['query_layer'], input_dict['key_layer'].transpose(-1, -2))
and then I check the above hlo 

```
HloModule SyncTensorsGraph.11, is_scheduled=true, entry_computation_layout={(bf16[2,32,4096,128]{3,2,1,0}, bf16[2,32,4096,128]{3,2,1,0})->(bf16[2,32,4096,4096]{3,2,1,0})}, replica_count=8, frontend_attributes={fingerprint_before_lhs="2cd58f17c9bec52ece9c4efe567dcd2c"}

wrapped_transpose_computation {
  param_0 = bf16[2,32,4096,128]{3,2,1,0} parameter(0)
  ROOT transpose.1.1 = bf16[2,32,128,4096]{3,2,1,0} transpose(param_0), dimensions={0,1,3,2}, metadata={op_type="aten__permute" op_name="aten__permute" source_file="test.py" source_line=5}
}

ENTRY SyncTensorsGraph.11 {
  p1.5.0 = bf16[2,32,4096,128]{3,2,1,0} parameter(1), metadata={op_type="xla__device_data" op_name="xla__device_data" source_file="test.py" source_line=5}
  p0.1.0 = bf16[2,32,4096,128]{3,2,1,0} parameter(0), metadata={op_type="xla__device_data" op_name="xla__device_data" source_file="test.py" source_line=5}
  bitcast.26.0 = bf16[64,4096,128]{2,1,0} bitcast(p1.5.0)
  wrapped_transpose = bf16[2,32,128,4096]{3,2,1,0} fusion(p0.1.0), kind=kInput, calls=wrapped_transpose_computation, metadata={op_type="aten__permute" op_name="aten__permute" source_file="test.py" source_line=5}
  bitcast.31.0 = bf16[64,128,4096]{2,1,0} bitcast(wrapped_transpose)
  custom-call.1.0 = (bf16[64,4096,4096]{2,1,0}, s8[33554432]{0}) custom-call(bitcast.26.0, bitcast.31.0), custom_call_target="__cublas$gemm", metadata={op_type="aten__matmul" op_name="aten__matmul" source_file="test.py" source_line=5}, backend_config={"operation_queue_id":"0","wait_on_operation_queues":[],"gemm_backend_config":{"alpha_real":1,"alpha_imag":0,"beta":0,"dot_dimension_numbers":{"lhs_contracting_dimensions":["2"],"rhs_contracting_dimensions":["1"],"lhs_batch_dimensions":["0"],"rhs_batch_dimensions":["0"]},"precision_config":{"operand_precision":["DEFAULT","DEFAULT"],"algorithm":"ALG_UNSET"},"epilogue":"DEFAULT","damax_output":false,"lhs_stride":"524288","rhs_stride":"524288","grad_x":false,"grad_y":false},"force_earliest_schedule":false}
  get-tuple-element.1 = bf16[64,4096,4096]{2,1,0} get-tuple-element(custom-call.1.0), index=0, metadata={op_type="aten__matmul" op_name="aten__matmul" source_file="test.py" source_line=5}
  bitcast.35.0 = bf16[2,32,4096,4096]{3,2,1,0} bitcast(get-tuple-element.1)
  ROOT tuple.10.0 = (bf16[2,32,4096,4096]{3,2,1,0}) tuple(bitcast.35.0)
} // SyncTensorsGraph.11
```
But I didn't find anything unusual, can you provide some help?
